src/optimizer/pushdown/pushdown_mark_join.cpp

```diff
@@ -9,14 +9,16 @@ using Filter = FilterPushdown::Filter;
 unique_ptr<LogicalOperator> FilterPushdown::PushdownMarkJoin(unique_ptr<LogicalOperator> op,
                                                              unordered_set<idx_t> &left_bindings,
                                                              unordered_set<idx_t> &right_bindings) {
+	auto op_bindings = op->GetColumnBindings();
 	auto &join = op->Cast<LogicalJoin>();
 	auto &comp_join = op->Cast<LogicalComparisonJoin>();
 	D_ASSERT(join.join_type == JoinType::MARK);
 	D_ASSERT(op->type == LogicalOperatorType::LOGICAL_COMPARISON_JOIN ||
 	         op->type == LogicalOperatorType::LOGICAL_DELIM_JOIN || op->type == LogicalOperatorType::LOGICAL_ASOF_JOIN);
 
 	right_bindings.insert(comp_join.mark_index);
-	FilterPushdown left_pushdown(optimizer, convert_mark_joins), right_pushdown(optimizer, convert_mark_joins);
+	FilterPushdown left_pushdown(optimizer, projected_mark_indexes, convert_mark_joins),
```

Comment:
I'm wondering if the `FilterPushdown` optimizer is the right place to do this optimization, given that we are now adding even more new parameters here only to facilitate this one optimization (that actually has little to do with filter pushdown). 

Perhaps we should move this to a separate optimizer altogether so that we can clean up the filter pushdown optimizer and leave it to only push down filters?

I think it was originally added there because it seemed easy but given that this has caused a number of issues it seems that was the wrong assumption.